content/ref/python/python_api_walkthrough.md

```diff
@@ -0,0 +1,267 @@
+print("Target path: ", target_path)
+
+run = wandb.init(entity=TEAM_ENTITY, project=PROJECT)
+model_artifact = run.use_artifact(artifact_or_name=artifact_name, type="model")
+run.link_artifact(artifact=model_artifact, target_path=target_path)
+run.finish()
+```
+
+Within the Registry App, you should see the model artifact within the DemoModels collection in the Model registry. You can click on it to view its details, including the version history, [lineage map]({{< relref path="../../guides/core/registry/lineage/" >}}), and other [metadata]({{< relref path="../../guides/core/registry/registry_cards/" >}}). 
+
+For additional information on how to link artifacts to a registry, see [Link artifacts to a registry]({{< relref path="../../guides/core/registry/link_version/" >}}).
+
+## Retrieve model artifact from registry for inference
+
+To use a model for inference, use `wandb.use_artifact` to retrieve the published artifact from the Model registry. This returns an artifact object that you can then use the [`Artifact.download`]({{< relref path="./artifact.md#download" >}}) method to download the artifact.
+
+```python
+REGISTRY_NAME = "Model"  # Name of the registry in W&B
+COLLECTION_NAME = "DemoModels"  # Name of the collection in the registry
+VERSION = 0 # Version of the artifact to retrieve
+
+model_artifact_name = f"wandb-registry-{REGISTRY_NAME}/{COLLECTION_NAME}:v{VERSION}"
+print(f"Model artifact name: {model_artifact_name}")
+
+run = wandb.init(entity=TEAM_ENTITY, project=PROJECT)
+registry_model = run.use_artifact(artifact_or_name=model_artifact_name)
+local_model_path = registry_model.download()
+```
+
+For more information on how to retrieve artifacts from a registry, see [Download an artifact from a registry]({{< relref path="../../guides/core/registry/download_use_artifact/" >}}).
+
+Depending on your machine learning framework, you may need to recreate the model architecture before loading the weights. This is left as an exercise for the reader, as it depends on the specific framework and model you are using. 
+
+## Share your finds with a report
+
+Create and share a [report]({{< relref path="../../guides/core/reports/_index.md" >}}) to summarize your work. To create a report programmatically, use the [W&B Reports API]({{< relref path="./wandb_workspaces/reports.md" >}}).
+
+First, install the W&B Reports API:
+
+
+```python
+!pip install wandb wandb-workspaces -qqq
+```
+
+Next, import the package:
+
+
+```python
+import wandb_workspaces.reports.v2 as wr
+```
+
+The following code block creates a report with multiple blocks, including markdown, panel grids, and more. You can customize the report by adding more blocks or changing the content of existing blocks. 
+
+The output of the code block prints a link to the URL report created. You can open this link in your browser to view the report. 
+
+
+```python
+experiment_summary = """This is a summary of the experiment conducted to train a simple model using W&B."""
```

Comment:
```suggestion
```python
import wandb_workspaces.reports.v2 as wr

experiment_summary = """This is a summary of the experiment conducted to train a simple model using W&B."""
```

Reply:
I left this as a separate step because I thought it might highlight that we are using a different library. If you don't think that'll do the job, I can update it to ^^

Reply:
I just think in terms of UX the "copy/paste-ability" is enhanced by keeping the code block contiguous; you can use a comment for highlighting the library name if it's sticking in your craw still but I think the fact that it's in the code as-is and will break if you change it will also help keep people on the path. :)

Reply:
Good point!